Revive: Parallel replicas custom key: skip unavailable replicas

devcrafter opened the pull request:
Revive https://github.com/ClickHouse/ClickHouse/pull/57235 - Parallel replicas custom key: skip unavailable replicas. Reverts ClickHouse/ClickHouse#58992

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

devcrafter committed:
Revert "Revert flaky"

robot-ch-test-poll2 commented:
<!-- automatic status comment for PR #58993 from ClickHouse/ClickHouse:revert-58992-revert_flaky -->
*This is an automated comment for commit 131f71b50aedabbc913443747467cecd3fda8778 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/58993/131f71b50aedabbc913443747467cecd3fda8778/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/58993/131f71b50aedabbc913443747467cecd3fda8778/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

devcrafter committed:
Disable check which is flaky atm

devcrafter commented:
`tests/integration/test_parallel_replicas_custom_key_load_balancing/test.py` is checking that behavior is the same as before, i.e. when all nodes available, - load balancing spreads load evenly between different nodes

devcrafter committed:
Update tests

+ setting max_replica_delay_for_distributed_queries to avoid replica lag
  affecting load balancing
+ disable flaky assert for asan/tsan builds (will investigated
  separately)

robot-clickhouse committed:
Automatic style fix

devcrafter committed:
Comment for disabled assert

devcrafter committed:
Merge remote-tracking branch 'origin/master' into revert-58992-revert_flaky
